File: frontend/src/types/deletionLogs.ts
export type DeletionLogStatus = "success" | "failed";

export type DeletionLogCountKey =
  | "deleted_chunks"
  | "deleted_entities"
  | "deleted_relationships"
  | "deleted_agent_runs"
  | "deleted_agent_steps"
  | "deleted_chat_messages"
  | "deleted_chat_sessions";

export type DeletionLog = {
  id: string;
  document_id: string;
  file_name: string | null;
  status: DeletionLogStatus;
  failure_stage: string | null;
  error_message: string | null;
  deleted_storage_file: boolean;
  deleted_qdrant_points: boolean;
  deleted_chunks: number;
  deleted_entities: number;
  deleted_relationships: number;
  deleted_agent_runs: number;
  deleted_agent_steps: number;
  deleted_chat_messages: number;
  deleted_chat_sessions: number;
  created_at: string;
};

export type DeletionLogListResponse = {
  logs: DeletionLog[];
  limit: number;
  offset: number;
  has_more: boolean;
};

File: backend/app/schemas/deletion_logs.py
from datetime import datetime
from typing import Annotated, Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field, StrictBool, StrictInt


NonNegativeInt = Annotated[StrictInt, Field(ge=0)]
DeletionLogStatus = Literal["success", "failed"]


class DeletionLogResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: UUID
    document_id: UUID
    file_name: str | None
    status: DeletionLogStatus
    failure_stage: str | None
    error_message: str | None
    deleted_storage_file: StrictBool
    deleted_qdrant_points: StrictBool
    deleted_chunks: NonNegativeInt
    deleted_entities: NonNegativeInt
    deleted_relationships: NonNegativeInt
    deleted_agent_runs: NonNegativeInt
    deleted_agent_steps: NonNegativeInt
    deleted_chat_messages: NonNegativeInt
    deleted_chat_sessions: NonNegativeInt
    created_at: datetime


class DeletionLogListResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    logs: list[DeletionLogResponse]
    limit: int = Field(ge=1, le=100)
    offset: int = Field(ge=0)
    has_more: StrictBool


__all__ = [
    "DeletionLogListResponse",
    "DeletionLogResponse",
    "DeletionLogStatus",
]
